ls1intum/Artemis
# [Bugfix] update Gitlab members only if they exist

### Checklist

### Motivation and Context


This PR should fix #2772.

Gitlab returns an error status code when attempting to add a member to a repository that already exists. Therefore, exceptions occur when calling `addMemberToRepository` in `GitlabService` for existing members. This happens e.g when attempting to lock/unlock repositories for a programming exercise.

### Description

We add a new check in `addMemberToRepository`: We check if the member already exists in the repository. If it does, we simply update its permissions. If it doesn't, we add it to the repository.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration, create a programming exercise, and participate in it by creating a submissions
4. Navigate to Course Administration -> Edit Exercise
5. Press Lock All Repositories button
6. Navigate to the Gitlab repository and click on "Members"
7. Check that the student that submitted the code has "Guest" access level
8. Press Lock All Repositories again and double check that the access level stays as "Guest"
9. Press Unlock All Repositories
10. The member should now have access level "Developer"
11. Press Unlock All Repositories again and double check that the access level stays the same

### Test Coverage
